sound/soc/intel/boards/hda_dsp_common.c

```diff
@@ -54,6 +54,11 @@ int hda_dsp_hdmi_build_controls(struct snd_soc_card *card,
 	hda_pvt = snd_soc_component_get_drvdata(comp);
 	hcodec = &hda_pvt->codec;
 
+	if (hda_pvt->hdac_common.driver_type != HDAC_HDA) {
```

Comment:
wouldn't it be possible to check the component driver, maybe by checking its name? That would avoid adding more ad-hoc fields and would be more explicit IMHO.

Reply:
@lyakh Agreed, this would be better. I've tried to dig this info from existing structs, but at least in the ASoC component driver data, the information same for both. Let me check still the path from asoc component -> struct device -> driver. I think I tried that already, but it didn't work for some reason.

Reply:
but what about `comp->driver`?

Reply:
@lyakh that doesn't have the name, but "comp->dev->driver->name" could be used. But as hdac_hda.c is a "proxy driver" (this is used to expose all the dozens of HDA codecs drivers in ASoC framework, as ASoC codec drivers), this field doesn't get you "hdac_hda.c" (the ASoc driver), but instead the wrapperd driver. But hmm, for this specific purpose this would do. We've had a few other cases (like setting the autosuspend), where we'd really want the info whether the "hdac_hda.c" wrapper is used or not... I'll check them as well.

Reply:
`struct snd_soc_component_driver` has a `.name` field, or do you mean, that neither `hdmi_hda_codec` nor `hdac_hda_codec` set it? Maybe we could add names there? Or does `comp->driver ` point to other drivers?

Reply:
@lyakh I did a version where I indeed added code to fill the .name field in hdac-hda.c and hdac-hdmi.c and then you could implement this in a nicer fashion. But, but, I now digged deeper into drivers/base/ code to understand what happens in the case, and ended up making a patch for the rootcause. With the v3 fix, we'll just never hit this out-of-bounds issue. Adding the .name fields might still be needed for other purposes, but I'll do that in a separate patch.